BTree.cc

```diff
@@ -18,26 +19,599 @@ using std::cout;
+                        // left by one
+                        for ( int n = positionInParent; n < numKeysInParent-1; n++ ) {
+                            parent.setKey(n, parent.getKey(n+1));
+                            parent.setValue(n, parent.getValue(n+1));
+                            parent.setChild(n+1, parent.getChild(n+2));
+                        }
+
+                        // set the child in parent at the divider to the new block
+                        parent.setChild(positionInParent, numNewBlock);
+                    }
+
+                    // decrement number of keys in parent
+                    parent.setNumberOfKeys(numKeysInParent-1);
+
+                    // write newBlock to disk
+                    _file.putBlock(numNewBlock, newBlock);
+
+                    // dealoccate curr and sibling
+                    _file.deallocateBlock(numCurr);
+                    _file.deallocateBlock(numSibling);
+
+                    // write parent to disk
+                    _file.putBlock(numParent, parent);
+
+                } else {
+                    // we can remove a key from sibling, so we will
+                    // make it the divider in parent, and put the
+                    // divider from parent in curr
+                    if (hasLeft) {
+                        int positionInCurr = curr.getPosition(key);
+                        curr.insert(0, parent.getKey(positionInParent-1), parent.getValue(positionInParent-1), curr.getChild(positionInCurr));
+
+                        parent.setKey(positionInParent-1, sibling.getKey(numKeysInSibling-1));
+                        parent.setValue(positionInParent-1, sibling.getValue(numKeysInSibling-1));
+                        sibling.setNumberOfKeys(numKeysInSibling-1);
+                    } else {
+                        curr.insert(numKeysInCurr, parent.getKey(positionInParent), parent.getValue(positionInParent),sibling.getChild(0));
+
+                        parent.setKey(positionInParent, sibling.getKey(0));
+                        parent.setValue(positionInParent, sibling.getValue(0));
+
+                        // since we removed the first key in sibling,
+                        // we must slide the other keys in sibling left by one
+                        removeFirst(sibling);
+
+                    }
+
+                    _file.putBlock(numCurr, curr);
+                    _file.putBlock(numParent, parent);
+                    _file.putBlock(numSibling, sibling);
+
+
+                }
+                numCurr = path.top();
+                path.pop();
+                _file.getBlock(numCurr, curr);
+
+            }
+            if (isRoot(numCurr) && needsKeys(numCurr, curr)) {
+                // curr is the root and is empty
+
+                // set the root to the child of curr
+                _file.setRoot(numChild);
+
+                // deallocate curr
+                _file.deallocateBlock(numCurr);
+            }
+        }
+    }
```

Comment:
Need "return true;" on success.  Currently, it works, but the caller thinks it failed (and prints that in your removeresults.txt).